src/main/java/edu/kis/powp/jobs2d/command/ComplexCommand.java

```diff
@@ -18,6 +23,15 @@ public ComplexCommand(List<DriverCommand> listOfCommands, String name) {
 
     @Override
     public void execute(Job2dDriver driver) {
+        Dimension canvasDimension = DrawerFeature.getCanvasDrawAreaPanelController().getCanvasDimension();
+        CanvasBoundaryCheckVisitor boundaryCheckVisitor = new CanvasBoundaryCheckVisitor(canvasDimension);
+        this.accept(boundaryCheckVisitor);
+
+        if (boundaryCheckVisitor.doesExceedCanvas()) {
+            logger.warning("Drawing exceeds boundaries of canvas");
+            return;
+        }
```

Comment:
Not SRP. Do not add testing to ComplexCommand implementation. It is a good idea, but such feature should be provided in SRP&OCP way. In this case, maybe you can try converting to Decorator approach (in Job2dDriver or DriverCommand hierachy)

Reply:
I suggest providing a service that would make such a test every time the command is loaded. You can add the result to log or command window and implement the service as an observer to CommandManager.